typescript/headless/index.ts

```diff
@@ -1,15 +1,20 @@
-import api from "@flatfile/api";
-import { FlatfileEvent, FlatfileListener } from "@flatfile/listener";
+import api, { Flatfile } from "@flatfile/api";
+import { FlatfileListener } from "@flatfile/listener";
 import { automap } from "@flatfile/plugin-automap";
-import { FlatfileRecord, recordHook } from "@flatfile/plugin-record-hook";
+import { recordHook } from "@flatfile/plugin-record-hook";
 import { ExcelExtractor } from "@flatfile/plugin-xlsx-extractor";
 import nodemailer from "nodemailer";
 import { promisify } from "util";
 
+import fs from "fs";
+import path from "path";
+
 export default function flatfileEventListener(listener: FlatfileListener) {
   // 1.Create a Workbook
-  listener.on("space:created", async (event: FlatfileEvent) => {
-    const { spaceId, environmentId } = event.context;
+
+  let workbook: Flatfile.WorkbookResponse;
```

Comment:
Not in love with this method, would like to only have workbook accessed when needed, but I couldn't find a way to get it in the mapping portion of the code, as it all pointed to the workbook created for the file instead of the broader workbook that was being worked on.